### ISSUE TITLE
Streamlining selection criteria in terminology construction

### ISSUE THREAD

**Issue by Ca5e** [tno-terminology-design/tev2-specifications#15]
Multiple keywords are described within the current [terminology construction](https://tno-terminology-design.github.io/tev2-specifications/docs/spec-tools/terminology-construction) specifications to select terms using certain criteria. An example being the following.
```yaml
- "tags[management]@essif-lab" # import all terms from the mrg of `essif-lab:latest` that have grouptag `management`
- "termids[party,community]@essif-lab:0.9.4" # import the terms `party` and `community` from the mrg of `essif-lab:0.9.4`
- "-tags[terminology]" # remove all terms tagged with the grouptag `terminology`
- "rename party term: partij, glossaryTerm: Partij" # rename the term `party` into `partij`, including its `glossaryTerm` key
- "*@tev2" # import all terms defined in the scope `tev2`
```
To streamline the current specifications slightly and expand on the possibilities of the term selection criteria, the following syntax may be of use. This enables the specific syntax for `tags` and `termids` to be removed. Separation of any `key` and `rename` specifically can be done by matching `rename` as a rename task only if it the "rename" text is trailed by a space character.
```yaml
- "key[value, value]@scopetag:versiontag" # add terms within MRG `scopetag:versiontag` that have any of `value` as `key`
- "-key[value, value]" # remove all terms that have any of `value` as `key`
```
If the value found in an entry, that corresponds with the key, is of type list: if the list includes any of the specified values, it is considered a matching entry.
The abovementioned new syntax may have to be changed to accommodate values that include the value separator (a comma), possibly by surrounding the value with quotes.

**Comment by RieksJ** [comment]
@Ca5e can you review the following [specifications](https://tno-terminology-design.github.io/tev2-specifications/docs/spec-tools/terminology-construction) for terminology construction, and post your findings in a comment here below?

**Comment by Ca5e** [comment]
I'll work on matching the MRGT with the new specifications soon. I've corrected a few small spelling issues, most of the info seems clear to me. The few questions that did pop up:

- Could you elaborate a bit more on the difference between `*` and `* @`? If `* @` is supposed to use all of the terms in the default version of the terminology of the current scope, creating an MRG with this termselcrit will have to halt until the default MRG is actually created, or should it attempt to read the MRG before it is created so it uses the last available version?
- It may also be wise to not use `latest` as an example for the version tag, descriptions like 'that are in the latest version' could cause some confusion as there is no actual "time based" latest.
On that previous note; looking at the examples under the header `<key> [ <value1>, <value2>, ... ] <source>`, some do imply that there is something like a latest terminology, I'm not sure if this is intentional.

**Comment by RieksJ** [comment]
@Ca5e :

The difference between `*` and `*@` is where the MRGT looks for stuff to include in the terminology-under-construction.
- `*` takes curated texts as the source of operation, whereas
- `*@` takes the MRG that contains the default version of the terminology of the current scope as the source of operation.

Since curated texts can be edited, the latter option allows curators to include both the edited version of a term, and its predecessor (of which it could rename, e.g., the 'term' field to distinguish between the two), in an MRG. This is explained at the bottom of the 'Examples' box. If you didn't see that, then where should I have put it so that you would have?

I'll update the text so that `latest` isn't used as a versiontag.

**Comment by RieksJ** [comment]
I've updated the specs for terminology construction, including the renaming things. I think the precision of specification has gone up, and I hope it is also clearer now to read.

However - you will need look at the sources, or wait until the error that the MRG-importer generated is resolved (which doesn't provide a message that helps me solve the problem). But since I've made a modification to the SAF, specifying both `essif-lab` and `essiflab` as scopetags, both of which have the same URL (which should be ok), and point to the same gitub repo (not quite sure that's ok though), it should be doable to fix something.

**Comment by Ca5e** [comment]
> Since curated texts can be edited, the latter option allows curators to include both the edited version of a term, and its predecessor (of which it could rename, e.g., the 'term' field to distinguish between the two), in an MRG. This is explained at the bottom of the 'Examples' box. If you didn't see that, then where should I have put it so that you would have?

I did catch the explanation box, but wanted to make sure I understood the specifications correctly before working on it. I believe I get it, so thanks for clearing up.

> However - you will need look at the sources, or wait until the error that the MRG-importer generated is resolved (which doesn't provide a message that helps me solve the problem).

The issue was caused by not providing the branch details of the scopedir in the url. This definitely points me to some improvements that should be made in the mrg-import tool. As of changing the url, the pipeline hangs on the vsntag missing from the essiflab MRG, the MRGT seems quite strict about needing that field. After that is changed, the MRGT will halt because of a technical issue where it can not iterate over an undefined object (altvsntags of the documentation version), as I seem to have missed one question mark in the code. A workaround I see with the current version of the tool would be to provide the altvsntags as something or an empty list.
I'll work on pushing fixes and improvements as soon as possible.

> specifying both essif-lab and essiflab as scopetags, both of which have the same URL (which should be ok)

This should indeed be fine, the `https://github.com/essif-lab/framework/docs` url or equivalent with `saf.yaml` added, as logical as it seems, just doesn't point to anything.
